### PR TITLE
DEPR: DatetimeArray/TimedeltaArray.__init__

### LINKED ISSUE #55623: DEPR: DTA/TDA.__init__
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
NA
```

### Issue Description

I'm finding myself having to fix bugs in these, but they don't really serve much purpose.  We should just use _from_sequence instead.  Would need to deprecate bc IIUC fastparquet uses these.

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #55623 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
looks good to me

File: doc/source/whatsnew/v2.2.0.rst [hunk 1]
```diff
@@ -478,6 +479,7 @@ Other Deprecations
 - Deprecated the extension test classes ``BaseNoReduceTests``, ``BaseBooleanReduceTests``, and ``BaseNumericReduceTests``, use ``BaseReduceTests`` instead (:issue:`54663`)
 - Deprecated the option ``mode.data_manager`` and the ``ArrayManager``; only the ``BlockManager`` will be available in future versions (:issue:`55043`)
 - Deprecated the previous implementation of :class:`DataFrame.stack`; specify ``future_stack=True`` to adopt the future version (:issue:`53515`)
+- Deprecating downcasting the results of :meth:`DataFrame.fillna`, :meth:`Series.fillna`, :meth:`DataFrame.ffill`, :meth:`Series.ffill`, :meth:`DataFrame.bfill`, :meth:`Series.bfill` in object-dtype cases. To opt in to the future version, use ``pd.set_option("future.no_silent_downcasting", True)`` (:issue:`54261`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.2.0.rst:482]: Is this from a follow up?

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.2.0.rst:482]: i think it snuck in during a merge, will remove

**Comment by jbrockmendel** [comment]
If there are no more comments, I plan to merge this in a week.

**Pull request merged**